src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+#include "common/JsonHelper.h"
+
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter) {
+        switch (*iter) {
+            case '\\': ss << "\\\\"; break;
+            case '"': ss << "\\\""; break;
+            case '/': ss << "\\/"; break;
+            case '\b': ss << "\\b"; break;
+            case '\f': ss << "\\f"; break;
+            case '\n': ss << "\\n"; break;
+            case '\r': ss << "\\r"; break;
+            case '\t': ss << "\\t"; break;
+            default: ss << *iter; break;
+      }
+  }
+  ss << '"';
+  return ss.str();
+}
```

Comment:
Had some time over awaiting a 'make test' ...

    /* ****************************************************************************
    *
    * stringReplace -
    *
    * NOTE
    * The caller of this function supplies an allocated buffer for the
    * output and it is the callers responsibility to make sure the out-buffer
    * is large enough.
    */
    char* stringReplace(const char* in, char* out)
    {
      int  inIx   = 0;
      int  outIx  = 1;

      out[0] = '"';

      while (in[inIx] != 0)
      {
        switch (in[inIx])
        {
        case '\\':
        case '"':
        case '/':
        case '\b':
        case '\f':
        case '\n':
        case '\r':
        case '\t':
            out[outIx++] = '\\';

        default:
            out[outIx++] = in[inIx++];
        }
      }

      out[outIx++] = '"';
      out[outIx]   = 0;

      return out;
    }